presto-spi/src/main/java/io/prestosql/spi/type/UnscaledDecimal128Arithmetic.java

```diff
@@ -728,6 +732,11 @@ public static boolean isStrictlyNegative(long rawLow, long rawHigh)
         return isNegative(rawLow, rawHigh) && (rawLow != 0 || unpackUnsignedLong(rawHigh) != 0);
     }
 
+    private static boolean isNegative(int lastRawHigh)
+    {
+        return lastRawHigh >>> 31 == 1;
```

Comment:
the one below uses `& SIGN_INT_MASK) != 0;`. I'm not sure which one is faster, but we could unify them one way or another

Reply:
This is extremely difficult to benchmark and might make a difference so I'd leave for now.

Reply:
please change it to `& SIGN_INT_MASK) != 0` to keep code consistent.